### PR TITLE
Appointment Resource action-comment extension and Appoint.comment patch operation

### PR DESCRIPTION
**Pull request by joe-explr**
Description
----
Appointment Resource action-comment extension and Appoint.comment patch operation.

PR Checklist
----
- [x] Screenshot(s) of changes attached before changes merged.
- [ ] Screenshot(s) of changes attached after changes merged and published.

image
image
image
image
image

### CODE REVIEW

**Review by SantanuBanerjee — APPROVED** [review]

**Review by BhavyaAcharya — APPROVED** [review]

File: content/millennium/r4/base/workflow/appointment.md [hunk 1]
```diff
@@ -51,14 +51,15 @@ The following fields are returned if valued:
 * [Comment](http://hl7.org/fhir/R4/appointment-definitions.html#Appointment.comment){:target="_blank"}
 * [Patient Instruction](http://hl7.org/fhir/R4/appointment-definitions.html#Appointment.patientInstruction){:target="_blank"}
 * [Requested period](http://hl7.org/fhir/R4/appointment-definitions.html#Appointment.requestedPeriod){:target="_blank"}
-* [Extensions including is cancelable, is reschedulable, group appointment id](#extensions){:target="_blank"}
+* [Extensions including is cancelable, is reschedulable, group appointment id, action comment](#extensions){:target="_blank"}
```
Comment (alexbeckner1) [inline content/millennium/r4/base/workflow/appointment.md:54]: can we list these in alphabetical order?

File: content/millennium/r4/base/workflow/appointment.md [hunk 2]
```diff
@@ -69,6 +70,7 @@ All URLs for custom extensions are defined as `https://fhir-ehr.cerner.com/r4/St
 
 ID                               | Value\[x] Type                                              | Description
 -----------------------------    |-------------------------------------------------------------|--------------------------------------------------------------------
+`action-comment`                 | [`string`](https://hl7.org/fhir/r4/datatypes.html#string)   | The comment to be associated with the action performed on the resource.
```
Comment (alexbeckner1) [inline content/millennium/r4/base/workflow/appointment.md:73]: can we list these in alphabetical order?

File: lib/resources/r4/appointment_patch.yaml [hunk 3]
```diff
@@ -271,3 +271,41 @@ operations:
...
+          "value": [
+            {
+              "url": "https://fhir-ehr.cerner.com/r4/StructureDefinition/action-comment",
+              "valueString": "not required, cancelled"
+            }
+          ]
+        }
+    note: >
+      <ul>
+        <li>This patch operation must be accompanied by the replace status patch operation when updating status to "arrived", "checked-in", "cancelled", and "fulfilled".</li>
```
Comment (alexbeckner1) [inline lib/resources/r4/appointment_patch.yaml:293]: This note is not accurate.

File: lib/resources/r4/appointment_patch.yaml [hunk 4]
```diff
@@ -271,3 +271,41 @@ operations:
...
+    description: Additional text to aid in facilitating the appointment.
+    example: |
+      {
+        "op": "replace",
+        "path": "/comment",
+        "value": "Arrived"
+      }
+    note: >
+      <ul>
+        <li>This Patch comment operation should be accompanied with replace slot operation with an appropriate Slot reference Id.</li>
```
Comment (alexbeckner1) [inline lib/resources/r4/appointment_patch.yaml:310]: "When present, this patch operation must be accompanied with replace slot Patch operation."

File: content/millennium/r4/base/workflow/appointment.md [hunk 5]
```diff
@@ -51,14 +51,15 @@ The following fields are returned if valued:
...
-* [Extensions including is cancelable, is reschedulable, group appointment id](#extensions){:target="_blank"}
+* [Extensions including is cancelable, is reschedulable, group appointment id, action comment](#extensions){:target="_blank"}
 
 ## Terminology Bindings
 
 <%= terminology_table(:appointment, :r4) %>
 
 ## Extensions
 
+* [Action Comment]
```
[the remark this answers is not in the document]
Comment (joe-explr) [reply content/millennium/r4/base/workflow/appointment.md:62]: Addressed in 4205ae8ab1907346bf0f704e3fd9134667ac0d9d

File: content/millennium/r4/base/workflow/appointment.md (hunk 2 above)
Comment (joe-explr) [reply content/millennium/r4/base/workflow/appointment.md:73]: Addressed in 4205ae8ab1907346bf0f704e3fd9134667ac0d9d

File: lib/resources/r4/appointment_patch.yaml (hunk 3 above)
Comment (joe-explr) [reply lib/resources/r4/appointment_patch.yaml]: Addressed in 36a457b6e11d570134c5c2ec0214aa0888c104cc

**Review by alexbeckner1 — APPROVED** [review]

**Review by SukeshKonjeti — APPROVED** [review]

**Pull request merged**